test/common/services/retry/retry.ts

```diff
@@ -17,56 +17,51 @@
  * under the License.
  */
 
-import { retryForTruthy } from './retry_for_truthy';
+import { FtrProviderContext } from '../../ftr_provider_context';
 import { retryForSuccess } from './retry_for_success';
+import { retryForTruthy } from './retry_for_truthy';
 
-export function RetryProvider({ getService }) {
+export function RetryProvider({ getService }: FtrProviderContext) {
   const config = getService('config');
   const log = getService('log');
 
   return new class Retry {
-    async tryForTime(timeout, block) {
+    public async tryForTime<T>(timeout: number, block: () => Promise<T>) {
       return await retryForSuccess(log, {
         timeout,
         methodName: 'retry.tryForTime',
-        block
+        block,
       });
     }
 
-    async try(block) {
+    public async try<T>(block: () => Promise<T>) {
```

Comment:
Yeah, I had thought about using `block: () => Promise<T> | T`, but that requires a lot of more complicated types and I can't think of any good reason to have a `retry.try()` call that isn't using an async function. Sure, `retry.try(() => 1)` is technically valid, even if it is pointless, but all you'd have to do to make the types work is `retry.try(async () => 1)`, so I'm going to leave it as is rather than make the types more complicated than they need to be.